src/main/resources/db/migration/0300/V333__ObservationPlotBiomassDetails.sql

```diff
@@ -0,0 +1,212 @@
+    small_trees_count_low INTEGER NOT NULL CHECK ( 0 <= small_trees_count_low),
+    small_trees_count_high INTEGER NOT NULL CHECK ( small_trees_count_low <= small_trees_count_high),
+    herbaceous_cover_percent FLOAT NOT NULL CHECK ( 0 <= herbaceous_cover_percent AND herbaceous_cover_percent <= 100),
+    soil_assessment TEXT NOT NULL,
+    water_depth_cm FLOAT CHECK ( 0 <= water_depth_cm),
+    salinity_ppt FLOAT CHECK ( 0 <= salinity_ppt),
+    ph FLOAT CHECK ( 0 <= ph AND ph <= 14),
+    tide_id INTEGER REFERENCES tracking.mangrove_tides,
+    tide_time TIMESTAMP WITH TIME ZONE,
+
+    CONSTRAINT mangrove_required_values
+        CHECK (
+            (forest_type_id = 1
+                 AND water_depth_cm IS NULL
+                 AND salinity_ppt IS NULL
+                 AND ph IS NULL
+                 AND tide_id IS NULL
+                 AND tide_time IS NULL) OR
+            (forest_type_id = 2
+                AND water_depth_cm IS NOT NULL
+                AND salinity_ppt IS NOT NULL
+                AND ph IS NOT NULL
+                AND tide_id IS NOT NULL
+                AND tide_time IS NOT NULL)
+            ),
+
+    PRIMARY KEY (observation_id, monitoring_plot_id),
+    FOREIGN KEY (observation_id, monitoring_plot_id)
+        REFERENCES tracking.observation_plots (observation_id, monitoring_plot_id)
+);
+
+CREATE INDEX ON tracking.observation_biomass_details(observation_id);
+CREATE INDEX ON tracking.observation_biomass_details(monitoring_plot_id);
+
+CREATE TABLE tracking.observation_biomass_quadrant_details (
+    observation_id BIGINT NOT NULL REFERENCES tracking.observations ON DELETE CASCADE,
+    monitoring_plot_id BIGINT NOT NULL REFERENCES tracking.monitoring_plots ON DELETE CASCADE,
+    position_id INTEGER NOT NULL REFERENCES tracking.observation_plot_positions,
+    description TEXT,
+
+    PRIMARY KEY (observation_id, monitoring_plot_id, position_id),
+    FOREIGN KEY (observation_id, monitoring_plot_id)
+        REFERENCES tracking.observation_biomass_details (observation_id, monitoring_plot_id)
+        ON DELETE CASCADE
+);
+
+CREATE UNIQUE INDEX ON tracking.observation_biomass_quadrant_details
+    (observation_id, monitoring_plot_id, position_id);
+
+CREATE INDEX ON tracking.observation_biomass_quadrant_details(observation_id);
+CREATE INDEX ON tracking.observation_biomass_quadrant_details(monitoring_plot_id);
+
+CREATE TABLE tracking.observation_biomass_quadrant_species (
+    observation_id BIGINT NOT NULL REFERENCES tracking.observations ON DELETE CASCADE,
+    monitoring_plot_id BIGINT NOT NULL REFERENCES tracking.monitoring_plots ON DELETE CASCADE,
+    position_id INTEGER NOT NULL REFERENCES tracking.observation_plot_positions,
+    species_id BIGINT REFERENCES species,
+    species_name TEXT,
+    is_invasive BOOLEAN NOT NULL,
+    is_threatened BOOLEAN NOT NULL,
+    abundance_percent FLOAT NOT NULL CHECK ( 0 <= abundance_percent AND abundance_percent <= 100),
```

Comment:
Is there a reason this is `FLOAT` instead of `NUMERIC`? We usually use `NUMERIC` because it doesn't introduce precision errors.